sdk/keyvault/azkeys/crypto/crypto_client.go

```diff
@@ -120,39 +121,49 @@ func NewClient(keyURL string, credential azcore.TokenCredential, options *Client
 // EncryptOptions contains optional parameters for Client.EncryptOptions
 type EncryptOptions struct {
 	// Additional data to authenticate but not encrypt/decrypt when using authenticated crypto algorithms.
-	AAD []byte `json:"aad,omitempty"`
+	AuthData []byte `json:"aad,omitempty"`
 
 	// Initialization vector for symmetric algorithms.
 	IV []byte `json:"iv,omitempty"`
 
 	// The tag to authenticate when performing decryption with an authenticated algorithm.
-	Tag []byte `json:"tag,omitempty"`
+	AuthTag []byte `json:"tag,omitempty"`
 }
 
-func (e EncryptOptions) toGeneratedKeyOperationsParameters(alg EncryptionAlgorithm, value []byte) generated.KeyOperationsParameters {
+func (e EncryptOptions) toGeneratedKeyOperationsParameters(alg EncryptionAlg, value []byte) generated.KeyOperationsParameters {
 	return generated.KeyOperationsParameters{
 		Algorithm: (*generated.JSONWebKeyEncryptionAlgorithm)(&alg),
 		Value:     value,
-		AAD:       e.AAD,
+		AAD:       e.AuthData,
 		Iv:        e.IV,
-		Tag:       e.Tag,
+		Tag:       e.AuthTag,
 	}
 }
 
 // EncryptResponse contains response fields for Client.EncryptResponse
 type EncryptResponse struct {
-	KeyOperationResult
+	// The algorithm used to encrypt
+	Algorithm *EncryptionAlg
+
+	AuthData []byte
+
+	AuthTag []byte
+
+	Ciphertext []byte
+
+	IV []byte
+
+	KeyID *string
 }
 
-func encryptResponseFromGenerated(i generated.KeyVaultClientEncryptResponse) EncryptResponse {
+func encryptResponseFromGenerated(i generated.KeyVaultClientEncryptResponse, alg EncryptionAlg) EncryptResponse {
 	return EncryptResponse{
-		KeyOperationResult: KeyOperationResult{
-			AdditionalAuthenticatedData: i.AdditionalAuthenticatedData,
-			AuthenticationTag:           i.AuthenticationTag,
-			IV:                          i.Iv,
-			ID:                          i.Kid,
-			Result:                      i.Result,
-		},
+		AuthData:   i.AdditionalAuthenticatedData,
+		AuthTag:    i.AuthenticationTag,
+		IV:         i.Iv,
+		KeyID:      i.Kid,
+		Ciphertext: i.Result,
+		Algorithm:  to.Ptr(alg),
```

Comment:
Should this also just be 'Alg'? I like it the way it is, but wasn't sure if it was supposed to match it.

Reply:
I'm going to leave it as Algorithm, it's consistent with other languages.